src/utils.zig

```diff
@@ -391,6 +396,10 @@ pub fn CircularBufferQueue(comptime T: type) type { // MARK: CircularBufferQueue
 		pub fn empty(self: *Self) bool {
 			return self.startIndex == self.endIndex;
 		}
+
+		pub fn full(self: *Self) bool {
```

Comment:
I would call this `atCapacity` or `reachedCapacity`. A resizeable data structure cannot be full.